PermissionsService/Services/PermissionsStore.cs

```diff
@@ -332,8 +334,10 @@ private async Task<string> FetchHttpSourceDocument(string sourceUri)
             }
             else
             {
-                var scopesByRequestUrl = new Dictionary<string, IEnumerable<ScopeInformation>>();
-                foreach (var request in requests)
+                var scopesByRequestUrl = new ConcurrentDictionary<string, IEnumerable<ScopeInformation>>();
+                var uniqueRequests = requests.DistinctBy(static x => $"{x.HttpMethod}{x.RequestUrl}", StringComparer.OrdinalIgnoreCase);
+
+                Parallel.ForEach(uniqueRequests, async request =>
```

Comment:
```suggestion
                ```suggestion
                await Parallel.ForEachAsync(uniqueRequests, async (request,_) =>
```
As the lambda passed is `async`